crates/pathfinder/src/state/sync/l2.rs

```diff
@@ -264,11 +264,14 @@ async fn download_block(
                     .with_context(move || format!("Verify block {}", block_number))?;
                 Ok((block, verify_result))
             });
+            #[allow(unused_variables)]
             let (block, verify_result) = verify_hash.await.context("Verify block hash")??;
-            if verify_result == VerifyResult::Mismatch {
-                let block_number = block.block_number;
-                tracing::warn!(?block_number, block_hash = ?expected_block_hash, "Block hash mismatch");
-            }
+            // FIXME: test block hashes aren't correct so this error breaks tests.
```

Comment:
I don't think that's necessarily an issue. Having correct block hashes in the test data would make it much harder to change the relevant part. And we have separate tests for `verify_hash` anyway, so in most tests it's irrelevant I think.